docs/contributing.md

```diff
@@ -39,36 +39,28 @@ To contribute code to Flax on GitHub, follow these steps:
 
 2. Install [Python >=3.7](https://www.python.org/downloads/).
 
-3. (Optional) Create a virtual environment or a Docker container. See
-   [`dev/README.md`](https://github.com/google/flax/blob/main/dev/README.md)
-   for details on how to set up a Docker Container. To set up a virtual environment,
-   run the following:
+3. Install the [uv](https://docs.astral.sh/uv/) package manager. This will help you to install
+   all the required packages.
 
    ```bash
-   python3 -m virtualenv env
-   . env/bin/activate
+   curl -LsSf https://astral.sh/uv/install.sh | sh
    ```
 
-   This ensures all your dependencies are installed in this environment.
-
 4. Clone your local forked Flax repo with `git clone`. Then, install the required packages
    with [PyPi](https://pip.pypa.io/en/stable/cli/pip_install/). This enables you to immediately
    test the code after modifying it:
 
    ```bash
    git clone https://github.com/YOUR_USERNAME/flax
    cd flax
-   pip install -e ".[all]"
-   pip install -e ".[testing]"
-   pip install -r docs/requirements.txt
+   uv sync --all-extras
```

Comment:
how does uv interact with virtual envs?

Reply:
I think maybe we should keep a few lines here about installing with pip for those who don't want to `curl | sh` a new binary.

Reply:
so there's a `uv venv` - we should _definitely_ keep encouraging people to use virtual-envs in our docs.